### PR TITLE
[libidn2] Fixes for mingw and minor issues

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  - [x] Fixes building with mingw (needed for curl). For faster build, it uses the cmake build system which is also used for MSVC.
  - [x] Installs the pc file also for cmake builds.
  - [x] Modernizes the portfile
  - [x] Use libunistring for non-windows

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes, but I am still working on this PR.

### CODE REVIEW

**Comment by dg0yt** [comment]
I can pick up libunistring in another PR. But this one is needed for updating curl.

File: ports/libidn2/portfile.cmake [hunk 1]
```diff
@@ -50,21 +51,21 @@ if (VCPKG_TARGET_IS_WINDOWS)
 else()
```
Comment (NancyLi1013) [inline ports/libidn2/portfile.cmake:51]: file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/libidn2/portfile.cmake [hunk 2]
```diff
@@ -9,62 +9,64 @@ vcpkg_download_distfile(ARCHIVE
 
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
+    ARCHIVE "${ARCHIVE}"
     REF ${IDN2_VERSION}
 )
 
 if (VCPKG_TARGET_IS_WINDOWS)
```
Comment (NancyLi1013) [inline ports/libidn2/portfile.cmake:16]:
```suggestion
if (VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_MINGW)
```
If using cmake to build on mingw, this condition is also required. I noticed that you add
```
{
      "name": "vcpkg-cmake",
      "host": true,
      "platform": "windows | mingw"
    }
```
in vcpkg.json file.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/libidn2/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/libidn2/portfile.cmake:16]: Note that the platform expression in the manifest is not consistent with the script variables:
https://github.com/microsoft/vcpkg/blob/cdd51899f6ae3736bfbe936f9e387f86d4a63a58/scripts/cmake/vcpkg_common_definitions.cmake#L52-L55

Comment (dg0yt) [reply ports/libidn2/portfile.cmake:16]: Verified again: Mingw uses cmake as expected. No change needed.

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**